python/pyspark/broadcast.py

```diff
@@ -177,28 +201,28 @@ def destroy(self, blocking=False):
         self._jbroadcast.destroy(blocking)
         os.unlink(self._path)
 
-    def __reduce__(self):
+    def __reduce__(self) -> Tuple[Callable[[int], T], Tuple[int]]:
         if self._jbroadcast is None:
             raise RuntimeError("Broadcast can only be serialized in driver")
-        self._pickle_registry.add(self)
-        return _from_id, (self._jbroadcast.id(),)
+        cast(Any, self._pickle_registry).add(self)
+        return cast(Tuple[Callable[[int], T], Tuple[int]], (_from_id, (self._jbroadcast.id(),)))
```

Comment:
I use `cast` to match with type hint of this functions

Reply:
https://github.com/apache/spark/pull/34439/files#r753855676